x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportFinalizeJobExecutionAction.java

```diff
@@ -57,6 +58,7 @@ protected void masterOperation(FinalizeJobExecutionAction.Request request, Clust
         clusterService.submitStateUpdateTask(source, new ClusterStateUpdateTask() {
             @Override
             public ClusterState execute(ClusterState currentState) throws Exception {
+                XPackPlugin.checkReadyForXPackCustomMetadata(currentState);
```

Comment:
@droberts195 this feels strange given the name of the method. If we get so far that `TransportFinalizeJobExecutionAction` is called, shouldn't the cluster state already have the ML type? can you please double check?

Reply:
Yes, I think the only endpoints that need protecting are the ones that put jobs and datafeeds.  Until the user has created an ML entity the endpoints that operate on them should be no-ops.

(There is actually a bug in this action in that if you passed it a non-existent job ID it would currently fail with an NPE.  That's not a disaster as it's an undocumented action intended for internal use, but I will make it more defensive in another PR.)